vxfreeswitch/client.py

```diff
@@ -0,0 +1,128 @@
+# -*- test-case-name: vxfreeswitch.tests.test_client -*-
+
+"""
+FreeSwitch ESL API client.
+"""
+
+from twisted.internet.protocol import ClientFactory
+from twisted.internet.defer import inlineCallbacks, returnValue, Deferred
+
+from eventsocket import EventProtocol, EventError
+
+
+class FreeSwitchClientProtocol(EventProtocol):
+    """ Freeswitch ESL client.
+
+    :param str auth:
+        Authentication string to send to FreeSwitch.
+    """
+
+    def __init__(self, auth):
+        EventProtocol.__init__(self)
+        self._auth = auth
+        self._connected = Deferred()
+
+    @inlineCallbacks
+    def connectionMade(self):
+        if self._auth:
+            yield self.auth(self._auth)
```

Comment:
The example dialer has this in `authRequest` not `connectionMade`: https://github.com/fiorix/eventsocket/blob/master/examples/dialer.py#L24 . Which is the callback for all `auth/request` events. Do we not need to send it then, or does freeswitch require the auth request to be sent as soon as we connect?